include/deal.II/matrix_free/matrix_free.h

```diff
@@ -630,6 +630,8 @@ class MatrixFree : public Subscriptor
          const std::vector<QuadratureType> &                    quad,
          const AdditionalData &additional_data = AdditionalData());
 
+  DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
```

Comment:
I am wondering if you can write:
```cpp
  template <typename QuadratureType, typename number2, typename DoFHandlerType>
  DEAL_II_DEPRECATED void
  reinit(const Mapping<DoFHandlerType::dimension> &                                   mapping,
         const std::vector<const DoFHandlerType *> &       dof_handler_hp,
         const std::vector<const AffineConstraints<number2> *> &constraint,
         const std::vector<QuadratureType> &                    quad,
         const AdditionalData &additional_data = AdditionalData());
```
The other function should be simply a specialization.

Reply:
...and leave out the extra diagnostics flag? Hmm this could work.

There are a couple of tests using the hp-variant in `tests/matrix_free`. I'll add a commit and we'll see whether the testsuite complains.